hedera-node/test-clients/scripts/diff-testing/gcp_access.py

```diff
@@ -0,0 +1,156 @@
+import os
+import re
+import warnings
+
+from google.cloud import storage
+from google.cloud.storage import transfer_manager
+from google.cloud.storage.retry import DEFAULT_RETRY
+
+from configuration import Configuration as C
+from globals import g
+from utils import split_list
+
+
+def init_gcp_and_get_streams_bucket():
+    """Initialize gcp library and return the interesting gcp bucket"""
+    g.gcp_retry_instance = DEFAULT_RETRY.with_delay(initial=C.GCP_RETRY_INITIAL_WAIT_TIME,
+                                                    multiplier=C.GCP_RETRY_WAIT_TIME_MULTIPLIER,
+                                                    maximum=C.GCP_RETRY_MAXIMUM_WAIT_TIME)
+
+    print(f"Getting gcp bucket {g.gcp_export_bucket}")
+
+    gcp_storage = storage.Client(project=g.gcp_project)
+    bucket = gcp_storage.bucket(g.gcp_export_bucket)  # this returns a Bucket you can access blobs with
+    return gcp_storage, bucket
+
+
+# Fetch file list
+
+def get_blobs_in_bucket(bucket, blob_glob, **kwargs):
+    """Given a blob glob - and optional lower and high blob names (w.r.t. timestamp) return all gcp Blobs that match"""
+    unambiguous_prefix = blob_glob.split('*')[0]
+    first_and_last_names = kwargs.get('first_and_last_names')
+    if first_and_last_names is None:
+        blobs = bucket.list_blobs(prefix=unambiguous_prefix, delimiter='/', match_glob=blob_glob,
+                                  timeout=C.GCP_SERVER_TIMEOUT, retry=g.gcp_retry_instance,
+                                  max_results=100000)
+    else:
+        blobs = bucket.list_blobs(prefix=unambiguous_prefix, delimiter='/', match_glob=blob_glob,
+                                  start_offset=first_and_last_names[0], end_offset=first_and_last_names[1],
+                                  timeout=C.GCP_SERVER_TIMEOUT, retry=g.gcp_retry_instance,
+                                  max_results=100000)
+    return [b for b in blobs]
+
+
+# Fetch file content
+
+def download_blobs(bucket, blobs_to_get, actually_present_blobs, a):
+    blob_list = sorted(blobs_to_get, key=lambda b: b.name)
+    # get batches
+    print(f"download_blobs: a = {a}")
```

Comment:
What is "a" here?

Reply:
My fault, I should have commented somewhere.

All of the command-line args - all of them, even the ones that you didn't specify which all have default values - are in a global variable `a` that is declared in `cli_arguments.py`.

(SImilarly there's a global `g` for actual "global variables" - I know that's bad software engineering but this is basically a grown-up script ... or at least that's my excuse - and that's declared in `globals.py`)

Reply:
😂 sounds good 👍